python/graphstorm/model/utils.py

```diff
@@ -703,8 +703,85 @@ def shuffle_predict(predictions, id_mapping_file, pred_type,
                 len(predictions)).cpu() # predictions are stored in CPU
     return predictions[local_id_mapping]
 
+def shuffle_nids(g, ntype, nids, node_id_mapping_file, rank):
+    """ Shuffle node ids into the original node ids according to node_id_mappings
+
+        Parameters
+        ----------
+        g: Dist DGLGraph
+            Graph.
+        ntype: str
+            Node type of nids.
+        nids: torch.Tensor
+            Node ids.
+        node_id_mapping_file:
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        rank: int
+            Rank of the current process in a distributed environment.
+    """
+    num_nodes = g.num_nodes(ntype)
+    id_mapping_info = create_dist_tensor((num_nodes,), dtype=th.int64,
+                                          name=f"mapping-{ntype}",
+                                          part_policy=g.get_node_partition_policy(ntype),
+                                          # TODO: this makes the tensor persistent in memory.
+                                          persistent=True)
+    if rank == 0:
+        id_mappings = th.load(node_id_mapping_file)
+        # the id_mapping stores the mapping from shuffled node ID to original ID
+        # For example, the id_mapping [1, 0 ,2] means:
+        # Original node ID: 1, 0, 2
+        # Shuffled node ID: 0, 1, 2
+        id_mapping = id_mappings[ntype] if isinstance(id_mappings, dict) else id_mappings
+        assert id_mapping.shape[0] == num_nodes, \
+            "id mapping should have the same size of num_nodes"
+        # Save ID mapping into dist tensor
+        id_mapping_info[th.arange(num_nodes)] = id_mapping
+    barrier()
+
+    return id_mapping_info[nids]
+
+def save_ep_prediction_results(predictions, src_nids, dst_nids,
```

Comment:
do you mean `save_edge_prediction_results`? ep is already edge prediction.

Reply:
Yes, this is save_edge_prediction. let me change the name.